src/analyze/analyze-security.c

```diff
@@ -0,0 +1,2038 @@
+
+                        r = table_update(details_table, cell, TABLE_STRING, buf);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to update cell in table: %m");
+                }
+
+                r = table_print(details_table, stdout);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to output table: %m");
+        }
+
+        exposure = DIV_ROUND_UP(badness_sum * 100U, weight_sum);
+
+        for (i = 0; i < ELEMENTSOF(badness_table); i++)
+                if (exposure >= badness_table[i].exposure)
+                        break;
+
+        assert(i < ELEMENTSOF(badness_table));
+
+        if (details_table) {
+                _cleanup_free_ char *clickable = NULL;
+                const char *name;
+
+                /* If we shall output the details table, also print the brief summary underneath */
+
+                if (info->fragment_path) {
+                        r = terminal_urlify_path(info->fragment_path, info->id, &clickable);
+                        if (r < 0)
+                                return log_oom();
+
+                        name = clickable;
+                } else
+                        name = info->id;
+
+                printf("\n%s %sOverall Exposure Level for %s%s: %s%" PRIu64 ".%" PRIu64 " %s%s %s\n",
+                       special_glyph(ARROW),
+                       ansi_highlight(),
+                       name,
+                       ansi_normal(),
+                       colors_enabled() ? strempty(badness_table[i].color) : "",
+                       exposure / 10, exposure % 10,
+                       badness_table[i].name,
+                       ansi_normal(),
+                       special_glyph(badness_table[i].smiley));
+        }
+
+        fflush(stdout);
+
+        if (overview_table) {
+                char buf[DECIMAL_STR_MAX(uint64_t) + 1 + DECIMAL_STR_MAX(uint64_t) + 1];
+                TableCell *cell;
+
+                r = table_add_cell(overview_table, &cell, TABLE_STRING, info->id);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add cell to table: %m");
+                if (info->fragment_path) {
+                        _cleanup_free_ char *url = NULL;
+
+                        r = file_url_from_path(info->fragment_path, &url);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to generate URL from path: %m");
+
+                        (void) table_set_url(overview_table, cell, url);
+                }
+
+                xsprintf(buf, "%" PRIu64 ".%" PRIu64, exposure / 10, exposure % 10);
+                r = table_add_cell(overview_table, &cell, TABLE_STRING, buf);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add cell to table: %m");
+                (void) table_set_align_percent(overview_table, cell, 100);
+
+                r
```

Comment:
The comments are not where they should be (i don't know why), this should not be an issue anyway; otherwise, say so.

Reply:
What I meant by `User=foo`, is not having `CAP_SYS_ADMIN`. See `context_has_no_new_privileges`, in `core/execute.c`: https://github.com/systemd/systemd/blob/master/src/core/execute.c#L1372

We need to set `NoNewPrivileges` in such case, otherwise seccomp will return `EACCES` ([man 2 seccomp](http://man7.org/linux/man-pages/man2/seccomp.2.html))

Reply:
I must be missing something, because in the code we are doing in that order: change UID (if `User=` is set), then we apply sandboxing features (including seccomp).
It means that by changing UID we are losing `CAP_SYS_ADMIN` (unless another option retains it), so `context_has_no_new_privileges` returns false.
Since it returns false, we set the `no_new_privileges` bit, otherwise `seccomp` returns `EACCES`.
All of that happens also in `--system` mode. Why do you say it only happens in `--user` mode?

Reply:
@ronnychevalier ah, true, we do set NNP for User= stuff in system mode too. Sorry for the confusion

Reply:
hmm, but we should really fix that, and keep CAP_SYS_ADMIN across the setresuid() transition, then do seccomp(), and then drop CAP_SYS_ADMIN again...

Reply:
But this would mean that we would not be able to blacklist `prctl` in the seccomp filters now, no?
What is the issue of enabling NNP in this case?

Reply:
I'll add a TODO list item for that

Reply:
> But this would mean that we would not be able to blacklist prctl in the seccomp filters now, no?

Probably, yes.

> What is the issue of enabling NNP in this case?

Well, it breaks setuid binaries, and I don't think we should do that unless we really have to.